subsys/bluetooth/host/conn.c

```diff
@@ -1008,7 +1008,7 @@ void bt_conn_security_changed(struct bt_conn *conn, enum bt_security_err err)
 		}
 	}
 #if (defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST))
-	if ((err == BT_SECURITY_ERR_SUCCESS) &&
```

Comment:
I think you need to squash this commit, the suggested changes doesn't work with zephyrs workflow.
Sorry about that.